src/lib/operators/maintenance/drop_table.cpp

```diff
@@ -14,15 +14,17 @@ const std::string DropTable::description(DescriptionMode description_mode) const
 }
 
 std::shared_ptr<const Table> DropTable::_on_execute() {
-  StorageManager::get().drop_table(table_name);
+  if (!if_exists || StorageManager::get().has_table(table_name)) {
```

Comment:
This could use a comment that we rely on the StorageManager to fail if the table does not exist but `IF EXISTS` is not set.